### ISSUE TITLE
I reproduced the code of pytorch version, but get different result

### ISSUE THREAD

**Issue by NostalgiaOfTime** [dandelin/ViLT#9]
In Image Retrieval, the R@1 is 68.4 which is higher than 61.9 in paper
In Text retrieval, the R@1 is   73.5 which is lower than 81.4 in paper

So, I want if the input format is error in my code.

In image, I use "pixelbert_transform" function of size=384. In Text, I use Bert base tokenizer with max len 40 which includes [CLS], word tokens and without [SEP]. In flickr-30k, I use dataset_flickr30k.json to get test datasets, and I chose the first caption of five about each image.

Thanks very much for your help!

**Comment by dandelin** [comment]
You mean you did not use the command in https://github.com/dandelin/ViLT/blob/master/EVAL.md#evaluate-f30k-irtr?

Tell me more details on your "reproducing the code of pytorch version", do you mean you pretrained a new ViLT from scratch?

The format you described seems good to me.

**Comment by NostalgiaOfTime** [comment]
Thanks for your reply!

It's not realistic for me to pretrain from scratch. I use the vilt_irtr_f30k.ckpt for testing the retrieval task. 
I reproduced the code of pytorch version for more flexible operation. Like Bert, which can extract features of the model, for example:

img_input = pixelbert_transform(img)
txt_input = tokenizer (text)
net = ViLT(config)

score= net(img = img_input, text=txt_input)

*************************************************************************
So, I just want if there any other specific operation in datasets when testing.

About image input. I use "pixelbert_transform" function of size=384
About text input, I use Bert base tokenizer with max len 40 which includes [CLS], word tokens and [SEP]
In flickr-30k, I use dataset_flickr30k.json to get test datasets, and I chose the first caption of five about each image.
And I remove any mask code just for testing.
In retrieval, I use best score of rank_out as R@1
[vilt_module.zip](https://github.com/dandelin/ViLT/files/6620878/vilt_module.zip)

I think the problem occurs in input format or datasets.

 Thanks for your help!

**Comment by dandelin** [comment]
Going over your description, I suspect the difference is from the way to compute recalls.

As you know, each image in COCO and F30K has five captions on average.
For the image retrieval, R@1 can be measured easily since there is only one image mapped queried text.
However, text retrieval R@1 is measured like: "does one of my five captions has retrieved as top-1 similar caption?" which is somewhat easier than the image retrieval.
Since you discarded the other 4 captions from the evaluation, I feel it is natural that your get a higher score in image retrieval and a lower score in text retrieval.
I refer you to [our code computing the recalls](https://github.com/dandelin/ViLT/blob/master/vilt/modules/objectives.py#L445).

**Comment by NostalgiaOfTime** [comment]
Thanks for your careful help, and I get the same result.
